src/core/server/http/http_service.ts

```diff
@@ -69,7 +73,8 @@ export class HttpService implements CoreService<HttpServiceSetup, HttpServiceSta
 
     const config = await this.config$.pipe(first()).toPromise();
 
-    return this.httpServer.setup(config);
+    const setup = this.httpServer.setup(config);
+    return { ...setup, ...{ createNewServer: (cfg: HttpConfig) => this.createServer(cfg) } };
```

Comment:
now `createNewServer` is not included in docs, because `HttpServiceSetup` refers to the server `setup` only
https://github.com/elastic/kibana/pull/36804/files#diff-f6edd5d99b796ea3adfdd32e2291ca1eR32
should be
```typescript
export type HttpServiceSetup = HttpServerSetup & {
  createNewServer: (cfg: HttpConfig) => Promise <HttpServer>
}
```

Reply:
ooh thank you